File: 1.TwoSum.java
/**
Given nums = [2, 7, 11, 15], target = 9,

Because nums[0] + nums[1] = 2 + 7 = 9,
return [0, 1].
*/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i, j;
        int[] result = new int[2];
        for(i = 0; i < nums.length; i++) {
            for(j = i+1; j < nums.length; j++) {
                if(nums[i] + nums[j] == target) {
                    result[0] = i;
                    result[1] = j;
                    return result;
                }
            }
        }
        result[0] = -1;
        result[1] = -1;
        return result;
    }
}

File: 1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Step 1: Initialize a dictionary to store numbers and their indices.
        num_to_index = {}
        
        # Step 2: Loop over each number in nums along with its index.
        for i, num in enumerate(nums):
            complementary = target - num
            if complementary in num_to_index:
                return [num_to_index[complementary], i]
            num_to_index[num] = i
        return []

# what is the output? 
# It returns a list, containing the indexes of numA and numB & numA + numB = target
# 
# are these numbers unique in the nums list? 
# no
